src/citrine/informatics/descriptors.py

```diff
@@ -82,6 +99,46 @@ def __repr__(self):
             self.key, self.lower_bound, self.upper_bound, self.units)
 
 
+class IntegerDescriptor(Serializable['IntegerDescriptor'], Descriptor):
+    """A descriptor to hold integer-valued numbers.
+
+    Parameters
+    ----------
+    key: str
+        the key corresponding to a descriptor
+    lower_bound: int
+        inclusive lower bound for valid integer values
+    upper_bound: int
+        inclusive upper bound for valid int values
+
+    """
+
+    lower_bound = properties.Integer('lower_bound')
+    upper_bound = properties.Integer('upper_bound')
+    units = properties.String('units', default='dimensionless')
+    typ = properties.String('type', default='Integer', deserializable=False)
+
+    def __eq__(self, other):
+        return self._equals(other, ["key", "lower_bound", "upper_bound", "units", "typ"])
+
+    def __init__(self,
+                 key: str,
+                 *,
+                 lower_bound: int,
+                 upper_bound: int):
+        self.key: str = key
+        self.lower_bound: int = lower_bound
+        self.upper_bound: int = upper_bound
+        self.units = "dimensionless"
```

Comment:
Should this be a default or just enforced as constant?

Reply:
I was on the fence about this one.  We could remove this line and the code will continue to work, due to our 'units' type hint which defaults to 'dimensionless'.

I'm not clear on how what you mean by enforce it as a constant.  Do you mean `self.unit = DIMENSIONLESS` which is defined elsewhere, e.g. the top of the file?  This could also be a class attribute, say `__DIMENSIONLESS` to discourage modification.

Reply:
Your approach would have done sufficed as well.  I meant mostly removing any route for a user to set the value (short of Python object hackery).

Reply:
Removing the instance attribute works fine, so I'll go with that approach.